SConstruct

```diff
@@ -7,8 +7,11 @@ import time
 import errno
 import SCons.Warnings
 from SCons.Script import BUILD_TARGETS
+from distutils.spawn import find_executable
 
 SCons.Warnings.warningAsException()
+# pylint: disable=undefined-variable
+# pylint: disable=unused-variable
```

Comment:
I think these could probably be fixed.

Reply:
Thanks for the review and comments.
I have removed both #pylint disable.

(1)From the SConstruct, the "Unused-variable" is used by Export() argument as a 'String'. 
which causing error/warning by pylint.
 
394     CONF_DIR = ARGUMENTS.get('CONF_DIR', '$PREFIX/etc')
395
396     env.Alias('install', '$PREFIX')
397     platform_arm = is_platform_arm()
398     Export('DAOS_VERSION', 'API_VERSION',
399            'env', 'prereqs', 'platform_arm',
400            'CONF_DIR')

(2) Undefined-variable GetOption is defined on the run time.
...
 41 def run_checks(env):
 42     """Run all configure time checks"""
 43     if GetOption('help') or GetOption('clean'):
 44         return
 45     cenv = env.Clone()

Reply:
We have a "fake scons" for which the whole purpose is to appear similar to scons for linting, for GetOption() the fix is probably to add this export to fake scons.

If we're exporting a variable called platform_arm and it doesn't exist then that's a bug and we should just remove the export.

Reply:
This inspired me to take a look, and I've fixed a couple of these in https://github.com/daos-stack/daos/pull/4228